### ISSUE TITLE
Astro spread attributes are misparsed or fail with expression children

### ISSUE THREAD

**Issue by mkusaka** [markuplint/markuplint#3824]
- Markuplint version: `4.18.1`
- Parser lang: Astro (`@markuplint/astro-parser@4.18.0`)
- Node.js version: `24.12.0`
- OS: macOS 15.6

## Describe the bug

When linting Astro files with `@markuplint/astro-parser`, some Astro-valid spread attribute syntax is either converted into an incorrect AST or throws `ParserError`.

These examples are accepted by `astro-eslint-parser`.

This issue is only about spread attribute parser behavior. It is separate from the `<script>` diagnostic issue.

Minimal reproduction repository:

https://github.com/mkusaka/markuplint-astro-parser-error-repro

## Code Example or Playground URL

Each `.astro` file in the reproduction repository also contains comments with the individual command and representative error output.

References / evidence:

- Repro all-case runner: `scripts/run-cases.mjs`
- Repro parser comparison script: `scripts/inspect-cases.mjs`
- `pnpm inspect` compares `astro-eslint-parser` diagnostics with direct `@markuplint/astro-parser` results.

### Case 1: TypeScript assertion in spread attribute

```sh
pnpm case:ts-assertion-spread
```

```astro
<button type="button" {...{ command: 'close' } as any} commandfor="dialog-id">
  close
</button>
```

### Case 2: spread attribute + expression child

One smaller base pattern is a spread attribute and an expression child on the same element. It is not limited to dynamic tags, multiple spreads, conditional spreads, or `textarea`.

There is also a related component-wrapper pattern: a component has a spread attribute, and its children contain a descendant element with an Astro expression child. This matches the `EventPost` pattern from `apps/clinics-cloud/src/pages/seminar/[eventId].astro`.

```sh
pnpm case:static-spread-expression-child
pnpm case:component-spread-descendant-expression-child
pnpm case:conditional-spread
pnpm case:dynamic-tag-spread
pnpm case:dynamic-tag-multiple-spreads
```

```astro
<div {...props}>{label}</div>
```

```astro
<Comp {...enabled ? { title: title } : {}}>
  <div>{title}</div>
</Comp>
```

```astro
<textarea {...fieldSizingContent ? { 'data-field-sizing': 'content' } : {}}>{value}</textarea>
```

```astro
<ContainerTag class="container" {...containerProps}>{title}</ContainerTag>
```

```astro
<Element
  {...iconOnly ? { title: label } : {}}
  {...rest}
>
  {label}
</Element>
```

## Steps To Reproduce

1. Clone the reproduction repository.

   ```sh
   git clone https://github.com/mkusaka/markuplint-astro-parser-error-repro.git
   cd markuplint-astro-parser-error-repro
   ```

2. Install dependencies.

   ```sh
   pnpm install
   ```

3. Run the spread cases.

   ```sh
   pnpm case:ts-assertion-spread
   pnpm case:static-spread-expression-child
   pnpm case:component-spread-descendant-expression-child
   pnpm case:conditional-spread
   pnpm case:dynamic-tag-spread
   pnpm case:dynamic-tag-multiple-spreads
   ```

4. Compare `astro-eslint-parser` and `@markuplint/astro-parser` behavior directly.

   ```sh
   pnpm inspect
   ```

### Case 1: TypeScript assertion in spread attribute

`astro-eslint-parser` accepts this file, and `@markuplint/astro-parser` returns an AST. However, the spread attribute is truncated, and `as` / `any}` are treated as normal attributes.

```text
<markuplint> error: The "as" attribute is disallowed (invalid-attr)
<markuplint> error: The "any}" attribute is disallowed (invalid-attr)
```

`pnpm inspect` shows that `@markuplint/astro-parser` interprets the start tag like this:

```json
[
  { "type": "attr", "raw": "type=\"button\"", "name": "type" },
  { "type": "spread", "raw": "{...{ command: 'close' }" },
  { "type": "attr", "raw": "as", "name": "as" },
  { "type": "attr", "raw": "any}", "name": "any}" },
  { "type": "attr", "raw": "commandfor=\"dialog-id\"", "name": "commandfor" }
]
```

### Case 2: spread attribute + expression child

`astro-eslint-parser` accepts these files, but `@markuplint/astro-parser` throws `ParserError`.

```text
<markuplint> error: SyntaxError: Invalid tag syntax: "<div {...props}>{label}</div>" (parse-error)
```

```text
<markuplint> error: SyntaxError: Invalid tag syntax: "<Comp {...enabled ? { title: title } : {}}>
  <div>{title}</div>
</Comp>" (parse-error)
```

```text
<markuplint> error: SyntaxError: Invalid tag syntax: "<textarea {...fieldSizingContent ? { 'data-field-sizing': 'content' } : {}}>{value}</textarea>" (parse-error)
```

```text
<markuplint> error: SyntaxError: Invalid tag syntax: "<ContainerTag class="container" {...containerProps}>{title}</ContainerTag>" (parse-error)
```

```text
<markuplint> error: SyntaxError: Invalid tag syntax: "<Element
  {...iconOnly ? { title: label } : {}}
  {...rest}
>
  {label}
</Element>" (parse-error)
```

I also checked smaller variants locally:

- `<div {...props}>text</div>`: accepted
- `<div id="x">{label}</div>`: accepted
- `<div {...props}>{label}</div>`: `ParserError`
- `<Comp {...enabled ? { title: title } : {}}><div>{title}</div></Comp>`: `ParserError`

So this does not look limited to dynamic tags or conditional/multiple spreads. The failures include both "spread attribute and expression child on the same element" and "component spread attribute with a descendant expression child".

## The expected behavior

- `{...{ command: 'close' } as any}` should be treated as one spread attribute, not as a truncated spread plus bogus `as` / `any}` attributes.
- Astro-valid spread attribute patterns should be parsed, including static tags, dynamic tags, conditional spreads, multiple spreads, and component wrappers whose children contain expression children.

`astro-eslint-parser` accepts these examples. That makes this look like an issue in `@markuplint/astro-parser`'s conversion from Astro AST to markuplint AST, or in its tag/attribute parsing layer, rather than invalid Astro syntax.

**Comment by YusukeHirao** [comment]
Fixed via #3854 (v4) and mirrored on dev via #3857 (#3856).
